core-bundle/src/DependencyInjection/Compiler/DataContainerCallbackPass.php

```diff
@@ -54,26 +54,27 @@ private function getCallbacks(ContainerBuilder $container): array
                 $serviceId = (string) $container->getAlias($serviceId);
             }
 
+            $definition = $container->findDefinition($serviceId);
+            $definition->setPublic(true);
+
             foreach ($tags as $attributes) {
-                $this->addCallback($callbacks, $serviceId, $attributes);
+                $this->addCallback($callbacks, $serviceId, $definition->getClass(), $attributes);
```

Comment:
The method call should probably be moved outside the foreach loop.

Reply:
That's just a dumb property return method. I doubt there's any (performance) improvement on that?

Reply:
A variable can be optimized by the OPcache. But I checked this in VLD and it seems that the OPcache already does this optimization internally. So we do not have to change the code.